internal/types/keysign.go

```diff
@@ -6,48 +6,43 @@ import (
 	"fmt"
 
 	"github.com/hibiken/asynq"
-	"github.com/vultisig/mobile-tss-lib/tss"
 
+	keysignTypes "github.com/vultisig/commondata/go/vultisig/keysign/v1"
 	"github.com/vultisig/vultisigner/internal/tasks"
 )
 
-type KeysignRequest struct {
-	PublicKeyECDSA   string   `json:"public_key_ecdsa"`   // ECDSA public key, used to identify the backup file
-	Messages         []string `json:"messages"`           // Messages need to be signed
-	Session          string   `json:"session"`            // Session ID , it should be an UUID
-	HexEncryptionKey string   `json:"hex_encryption_key"` // Hex encryption key, used to encrypt the keysign messages
-	DerivePath       string   `json:"derive_path"`        // Derive Path
-	IsECDSA          bool     `json:"is_ecdsa"`           // indicate use ECDSA or EDDSA key to sign the messages
-	VaultPassword    string   `json:"vault_password"`     // password used to decrypt the vault file
+type KeysignAPIRequest struct {
+	PublicKey  string                       `json:"public_key"`  // public key, used to identify the backup file
+	Messages   []string                     `json:"messages"`    // Messages need to be signed
+	DerivePath string                       `json:"derive_path"` // Derive Path
+	IsECDSA    bool                         `json:"is_ecdsa"`    // indicate use ECDSA or EDDSA key to sign the messages
+	Payload    *keysignTypes.KeysignPayload `json:"payload"`     // keysign payload
```

Comment:
This is a misunderstanding , `Payload` should not exist in here ,  client side UI should construct the transaction , and then get the PreKeysignImage, the bytes get from wallet-core ,  and put it into `Messages` field 

No point of passing `Payload` to server side , as server will not able to parse it , and won't be able to construct the transaction either.